### ISSUE TITLE
Sync time with host via toolbox

### ISSUE THREAD

**Issue by fdawg4l** [vmware/vic#2972]
We need a way to sync time with the host and keep it in sync in the appliance with the ESX host.  We were using ntp (via the `timesync` daemon [which helpfully `systemd` decided we required]) until [this](https://github.com/vmware/vic/pull/2971) change.  The `timesync` daemon would slam the clock leading to sudden skew in the logs and weirdness with respect to deadline based timeouts (which use wallclock time).

Disabling `timesync` was the first fix, the next fix is to keep the vch in sync with the host over time.  Apparently the `open-vm-tools` toolbox uses a timesync message to keep this in sync via a background task.  We need to implement this piece in our toolbox to do the same.  If we don't the clock may continue to drift and eventually logs between the VCH and the ESX host won't line up (this is already a problem irrespective of the `timesync` service change).

**Comment by fdawg4l** [comment]
See #2340 for details on clock skew affecting deadline based timeouts (in the `context` package).

**Comment by fdawg4l** [comment]
Assigning to myself for now, but no reason this can't be rebalanced.

**Comment by fdawg4l** [comment]
cc @dougm, @hickeng, and @caglar10ur.

**Comment by hickeng** [comment]
@jzt This is the work I mentioned in the sprint demo needed for the logs to function as expected over time.
This work would almost certainly fall to Doug.

@dougm Is there someone you think would be good to train on the toolbox systems (not just the current code but how to determine implementation path for new function) so we don't have to always assign these to you?

**Comment by dougm** [comment]
@hickeng yes, this would be a good one start with as we already have the protocol handling in place for the Set_Option RPC, iirc that's where the time sync request comes in.  I could pair up with @jzt or whoever is interested.

**Comment by hickeng** [comment]
@dougm @jzt Sounds good - please consider this as high priority once the cp work is complete.

**Comment by jzt** [comment]
@dougm @hickeng sounds good, thanks.
